README.md

```diff
@@ -140,9 +140,135 @@ export function App() {
 }
 ```
 
+#### TypeScript Generics
+
+We also support the use of Generics to add type definitions for your props. If
+used, the props will be properly typed when the props are passed to the
+`updateWithProps` method.
+
+To utilise generics you can use one of two methods. In both of the exampled
+below, we create a custom internal type called `MySketchProps` which is a union
+type of `SketchProps` and a custom type which has a `rotation` key applied to
+it.
+
+> Sidenote:
+>
+> We could also write the `MySketchProps` type as an interface to do exactly the
+> same thing if that is to your personal preference:
+>
+> ```typescript
+> interface MySketchProps extends SketchProps {
+>   rotation: number;
+> }
+> ```
+
+This means, in these examples, that when the `rotation` prop that is provided as
+part of the `props` passed to the `updateWithProps` function, it will be
+correctly typed as a `number`.
+
+##### Usage with the `P5Instance` type
+
+```typescript
+import React, { useState, useEffect } from "react";
+import { ReactP5Wrapper, P5Instance, SketchProps } from "react-p5-wrapper";
+
+type MySketchProps = SketchProps & {
+  rotation: number;
+};
+
+function sketch<MySketchProps>(p5: P5Instance<MySketchProps>) {
```

Comment:
I think this should actually be:

```suggestion
function sketch(p5: P5Instance<MySketchProps>) {
```

Can you check as part of your tests and update if necessary @yevdyko?